autoemulate/preprocess_target.py

```diff
@@ -0,0 +1,661 @@
+import inspect
+
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.optim as optim
+from sklearn.base import BaseEstimator
+from sklearn.base import clone
+from sklearn.base import TransformerMixin
+from sklearn.compose import TransformedTargetRegressor
+from sklearn.decomposition import PCA
+from sklearn.pipeline import Pipeline
+from sklearn.utils.validation import check_is_fitted
+from torch.utils.data import DataLoader
+from torch.utils.data import TensorDataset
+
+
+def get_dim_reducer(
+    name,
+    reduced_dim=8,
+    hidden_layers=[64, 32],
+    epochs=2000,
+    batch_size=32,
+    learning_rate=1e-3,
+    beta=1.0,
+    verbose=False,
+):
+    """
+    Factory function to get a dimensionality reducer based on name.
+
+    Parameters
+    ----------
+    name : str or None
+        Name of the dimensionality reducer to use.
+        Options:
+        - 'PCA': Principal Component Analysis
+        - 'AE': Autoencoder
```

Comment:
I might have missed it but it looks like we don't have a non-variational autoencoder option? Btw AE is also mentioned in the 06_reaction_diffusion_time notebook (in section 2)
```suggestion

```